sql/core/src/main/scala/org/apache/spark/sql/SQLContext.scala

```diff
@@ -531,6 +531,32 @@ class SQLContext(@transient val sparkContext: SparkContext)
   }
 
   /**
+   * Applies a schema to an List of Java Beans.
+   *
+   * WARNING: Since there is no guaranteed ordering for fields in a Java Bean,
+   *          SELECT * queries will return the columns in an undefined order.
+   * @group dataframes
+   * @since 1.6.0
+   */
+  def createDataFrame(data: java.util.List[_], beanClass: Class[_]): DataFrame = {
+    val schema = getSchema(beanClass)
```

Comment:
Up to your judgment since you know the details better of course. If it's not worth the overhead to factor this out then leave it.